### PR TITLE
add eslint-rules

### LINKED ISSUE #4203: Forbid Array Destructuring in Params
We forbid assignments, but allow params. That's just as bad.

### PR DESCRIPTION
**Pull request by erwinmombay**
Closes  #4203

### CODE REVIEW

**Comment by erwinmombay** [comment]
@jridgewell does it make sense doing one-off files with singular rules or just create one amp-presubmit-checks file? (ones that we can rewrite anyways)

File: .eslintrc [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+  "rulePaths": ["build-system/eslint-rules/"],
+  "plugins": ["eslint-plugin-google-camelcase"],
   "ecmaFeatures": {
     "modules": true,
     "arrowFunctions": true,
     "blockBindings": true,
-    "forOf": false,
-    "destructuring": false,
-    "spread": false
+    "forOf": true,
```
Comment (jridgewell) [inline .eslintrc]: This'll include that huge Symbol polyfill.

**Comment by jridgewell** [comment]
> does it make sense doing one-off files with singular rules

👍

File: .eslintrc (hunk 1 above)
Comment (erwinmombay) [inline .eslintrc]: @jridgewell do we really care? since we don't ship the babel transpiled code

Comment (jridgewell) [inline .eslintrc]: Will the build step error if we include a for-of?

Comment (erwinmombay) [inline .eslintrc]: @jridgewell i don't believe so (not 100% sure, ill check), but it won't be transpiled in any way. ill go ahead and add a presubmit check for that too if we already don't have one

File: ads/google/a4a/utils.js [hunk 2]
```diff
@@ -137,8 +137,7 @@ function buildAdUrl(
   const viewportRect = viewportFor(global).getRect();
   const iframeDepth = iframeNestingDepth(global);
   const dtdParam = {name: 'dtd'};
-  const allQueryParams = [
```
Comment (erwinmombay) [inline ads/google/a4a/utils.js]: @bobcassels had to remove the spread operator as the code closure compiler generates isnt optimal. does the order matter here or can we just stick queryParams and unboundedQueryParams in the head of the array?

Comment (jridgewell) [inline ads/google/a4a/utils.js]: We can easily keep the order:

```js
const allParams = queryParams.concat(first, unboundedQueryParams, last)
```

**Comment by jridgewell** [comment]
Fixes https://github.com/ampproject/amphtml/issues/4203?

File: ads/google/a4a/utils.js (hunk 2 above)
Comment (erwinmombay) [inline ads/google/a4a/utils.js]: yeah was just curious, since i was too lazy lol

Comment (erwinmombay) [inline ads/google/a4a/utils.js]: implemented suggestion to not block review.

**Comment by erwinmombay** [comment]
@jridgewell PTAL

**Comment by jridgewell** [comment]
LGTM.

**Pull request merged**